pkg/stanza/operator/helper/emitter.go

```diff
@@ -145,9 +145,13 @@ func (e *LogEmitter) flusher(ctx context.Context) {
 		select {
 		case <-ticker.C:
 			if oldBatch := e.makeNewBatch(); len(oldBatch) > 0 {
-				e.flush(ctx, oldBatch)
+				e.flush(context.Background(), oldBatch)
 			}
 		case <-ctx.Done():
+			// flush currently batched entries
+			for oldBatch := e.makeNewBatch(); len(oldBatch) > 0; oldBatch = e.makeNewBatch() {
```

Comment:
Any specific scenario that warrants using a `for` instead of doing it just once? I would assume that at the point when `LogEmitter.Stop` is called the rest of the Stanza pipeline is stopped, so no new entries should be landing in the batch.

Not that the `for` hurts, it will work "more than" fine. :slightly_smiling_face:  I just wonder if I'm missing something.

Reply:
I think it's not necessary in practice, because other components shouldn't be calling `Process` after `Stop` was already called. And this is how the pipeline acts. @djaglowski am I wrong here?